amundsen_application/static/js/components/SearchPage/tests/index.spec.tsx

```diff
@@ -155,14 +210,84 @@ describe('SearchPage', () => {
     });
 
     afterAll(() => {
-      getSelectedTabByResourceTypeSpy.mockRestore();
       createSearchOptionsSpy.mockRestore();
     });
   });
 
+  describe('componentDidUpdate', () => {
+    let props;
+    let wrapper;
+    let searchAllSpy;
+    beforeAll(() => {
+      const setupResult = setup({
+        location: {
+          search: '/search?searchTerm=current&selectedTab=table&pageIndex=0', 
+          pathname: 'mockstr',
+          state: jest.fn(),
+          hash: 'mockstr',
+        }
+      });
+      props = setupResult.props;
+      wrapper = setupResult.wrapper;
+      const mockPrevProps = {
+        location: {
+          search: '/search?searchTerm=previous2&selectedTab=table&pageIndex=0', 
+          pathname: 'mockstr',
+          state: jest.fn(),
+          hash: 'mockstr',
+        }
+      };
+      searchAllSpy = jest.spyOn(props, 'searchAll');
+      wrapper.instance().componentDidUpdate(mockPrevProps);
+    });
+
+    it('calls setState', () => {
+      expect(setStateSpy).toHaveBeenCalledWith({ selectedTab: ResourceType.table });
+    });
+
+    it('calls searchAll', () => {
+      expect(searchAllSpy).toHaveBeenCalled();
+    });
+  });
+
+  describe('getSanitizedUrlParams', () => {
+    let props;
+    let wrapper;
+    let getSelectedTabByResourceTypeSpy;
+
+    beforeAll(() => {
+      const setupResult = setup({
+        location: {
+          search: '/search?searchTerm=current&selectedTab=table&pageIndex=0', 
+          pathname: 'mockstr',
+          state: jest.fn(),
+          hash: 'mockstr',
+        }
+      });
+      props = setupResult.props;
+      wrapper = setupResult.wrapper;
+      const mockResourceType = ResourceType.table;
+      getSelectedTabByResourceTypeSpy = jest.spyOn(wrapper.instance(), 'getSelectedTabByResourceType').mockImplementation(() => {
+        return mockResourceType;
+      });
+      
+      wrapper.instance().getSanitizedUrlParams('current', 0, ResourceType.table)
+    });
+
+    it('calls getSelectedTabByResourceType with correct value', () => {
+      expect(getSelectedTabByResourceTypeSpy).toHaveBeenCalledWith('table');
+    });
+
+    it('output of getSanitizedUrlParams is expected', () => {
```

Comment:
This test verifies the ideal case of `getSanitizedUrlParams`, which is that the searchTerm and index exist. To make the test of this method more robust, we should also set up the cases that would test the following https://github.com/lyft/amundsenfrontendlibrary/blob/b7486b019d303fa8e3d8ef7cdcc0618b036b314b/amundsen_application/static/js/components/SearchPage/index.tsx#L124-L125

which would involve adding a test like:
```
const expected = {'term': '', 'index': 0, 'currentTab': mockResourceType};
expect(wrapper.instance().getSanitizedUrlParams(undefined, undefined, ResourceType.table)).toEqual(expected);
```
nit: note the use of expecting `mockResourceType` simply because that is what the the mocked `getSelectedTabByResourceType` will return.